### PR TITLE
Bootstrap 4 Update + Upgrade

### PR DESCRIPTION
**Pull request by agessler**
@apepper FYI

### CODE REVIEW

**Comment by apepper** [comment]
FYI: After `npm install` `package-lock.json` is modified. Please change that.

**Comment by apepper** [comment]
Are the bootstrap fonts in https://github.com/Scrivito/scrivito_example_app_js/tree/master/src/assets/fonts/bootstrap still needed? If not, please remove them.

**Comment by apepper** [comment]
Search results are currently broken:

broken search-results

**Comment by apepper** [comment]
At the end of the landing page, there is not enough "dark" space:

end of landing space - to few space

**Comment by apepper** [comment]
Colomns on landing page are no longer centered. Is this because of a new setting?

col no longer centered

**Comment by apepper** [comment]
Some dates of blog entries are overlapping:

overlapping blog entries

**Comment by apepper** [comment]
On contact page: The contact form now has a border around it. Please remove the boarder.:

form with box

**Comment by apepper** [comment]
The icons of the carousel no longer match the wireframes or the previous version. This is how they previously looked: 
old carousel icon

**Comment by apepper** [comment]
Really nice work @agessler and @code-constructor !

**Comment by agessler** [comment]
@apepper /fonts/bootstrap are deleted... fontawesome could be deleted if we load them via CDN ... but the current solution vie locla font path is much more stable.

**Comment by agessler** [comment]
@apepper `Some dates of blog entries are overlapping...` for a better solution we need bigger markup changes

**Comment by apepper** [comment]
### Design feedback

/cc @agessler 

Really good work. I'm also quite impressed, how you did the js changes. Here is a list of some still open topics, which should be covered by the designer.

#### Tag-List 1 + 2 

If you make the window a bit smaller, a drop down arrow will appear, while no select box is visible:

tag dropdown is wrong

Also, if you search for "Berlin" and are in "table im hochformat" view mode, the tags are overlapping:

search results with table im hochformat

#### Arrow icons highlighted 1 + 2

If you click on on thumbnail gallery and then the "next" button, the button will still be highlighted, even after your done clicking:

our work click


The same is true, if you click on the "collapse" in "table im hochformat"  view port:

collapse after closing it - still highlighted


#### Blog entries

I already reported it, but I think this still should be fixed, even with more markup changes:

overlapping blog entries

**Comment by apepper** [comment]
I suggest, that @agessler takes care of the design feedback and @code-constructor of the code feedback.

**Comment by apepper** [comment]
I also saw, that the `TableWidget` did not get a rewrite with bootstrap 4. I suggest we does this as the next story (see [Trello](https://trello.com/c/QDn2gu5f/342-tablewidget-with-bootstrap-4)). This PR is already big enough.

**Comment by agessler** [comment]
@apepper 
Fixed issues above and please cut out the devider in blog timeline.
Just close `</ul>` and insert the devider in a new `<ul>`
```
<ul class="timeline">
  <li class="timeline-divider"><time datetime="2017-05">May 2017</time></li>
</ul>
```

File: src/Widgets/ThumbnailGalleryWidget/ThumbnailGalleryWidgetComponent.js [hunk 1]
```diff
@@ -83,8 +83,8 @@ class ThumbnailGalleryComponent extends React.Component {
           currentTag={ this.state.currentTag }
           setTag={ this.setTag }
         />
-        <div className="row">
-          <div className="gallery-box-wrapper">
+        <div className="">
```
[the remark this answers is not in the document]
Comment (agessler) [reply src/Widgets/ThumbnailGalleryWidget/ThumbnailGalleryWidgetComponent.js]: Just an artefact..

File: src/Widgets/BoxWidget/BoxWidgetEditingConfig.js [hunk 2]
```diff
@@ -5,6 +5,14 @@ Scrivito.provideEditingConfig('BoxWidget', {
   title: 'Box',
   thumbnail: `/${boxWidgetIcon}`,
   attributes: {
+    cardStyle: {
```
[the remark this answers is not in the document]
Comment (agessler) [reply src/Widgets/BoxWidget/BoxWidgetEditingConfig.js]: In this case its background color... but I think card style is more precise.. just my opinion.

**Comment by apepper** [comment]
FEEDBACK (1717c60)

Good job with the design @agessler. It now looks all really good!

As you might have already seen I added two commits: One for separating the month in the blog post lists and one to fix the author image. I can confirm, that the separation of months fixed the overlapping issue. Good job 🚀!

@code-constructor will take care of the code feedback, once he is back.

**Comment by apepper** [comment]
well done

Well done @agessler and @code-constructor. Thank you for working on this!

**Pull request merged**